src/nvim/misc1.c

```diff
@@ -2408,10 +2408,37 @@ get_number (
   if (msg_silent != 0)
     return 0;
 
+  Callback input_callback = { .type = kCallbackNone };
+  char_u *select;
+  int xp_type = EXPAND_NOTHING;
+  char *xp_arg = NULL;
+  int cmd_silent_save = cmd_silent;
+  const int save_ex_normal_busy = ex_normal_busy;
+
+  cmd_silent = false;  // Want to see the prompt.
+  cmdline_row = msg_row;
+
+  select = (char_u *)getcmdline_prompt('@', "", 0, xp_type, xp_arg,
```

Comment:
This puts the cursor in new line. How can we avoid that?

Reply:
@UtkarshMe The text that is to be displayed on the same line should be passed here as the second argument. You will need to add an argument to `get_number` so `prompt_for_number` and `ml_recover` can send this string in.